### DISCUSSION TITLE
Is there a Discord for Wakapi?

### DISCUSSION THREAD

**Discussion by technicallypete** [muety/wakapi#601] [Q&A]
Is there a Discord for Wakapi?

**Comment by muety** [comment]
Nope, there is only GitHub Discussions.

**Reply by technicallypete** [reply]
Does it make sense to have a channel? 2k stars and 130+ forks might be enough to warrant it for easier discussion and collab.

**Reply by muety** [reply]
No, I don't think it makes sense at this point. There is not much discussion going on right now, so I don't expect much more if we had a Discord. Despite that, not really sure what people would want to be talking about there.

**Reply by technicallypete** [reply]
I actually wanted to discuss having an api :)

I want to auto add a user while provisioning a server, but i have to use the html page routes, ie:

curl:

POST http://localhost:1337/signup
`location=America%2FNew_York&username=<username>&email=<email>&password=<password>&password_repeat=<password>`

~~POST http://localhost:1337/login~~
...

GET http://localhost:1337/summary

(and then scrape the API Key from the html and see the api key to the waka config) 


A larger discussion for this specific use case could even introduce a binary that just does this like

`$ wakapi user add ... > api_key.txt`

And use that for a sed command to add to the config, or pipe to another command like

`$ wakapi user add ... | wakapi config add --field api_key`

Or 

`$ wakapi user add ... && wakapi config add --field api_key --value $?`

(or return json and use jq on the output of the user add response, if returning more than just the api key string) 

Etc

In connection with the thread about Teams, I'd also want to create an initial "admin" or project manager role user (so if user exists, the binary can send error code 255 or whatever) 

And Id show the respective users their passwords that I generated for them in my own ui 

Anyway, Unless Im just completely missing an api that already allows this, and its something worth discussing, ill just open a new Issue

Thanks for all the work btw; This project and your ux seems to be designed very well!

**Reply by muety** [reply]
> Thanks for all the work btw; This project and your ux seems to be designed very well!

Glad to hear that, thanks! 

I created a new issue (#604) for your feature request. Will look into this hopefully soon.

There's also an issue (#411) for a stand-alone executable CLI tool to perform common Wakapi admin tasks. But tbh., it's probably not going to happen any time soon.

**Reply by technicallypete** [reply]
Awesome, thanks!!
